Python/Product/TestAdapter/TestContainerDiscoverer.cs

```diff
@@ -114,18 +116,18 @@ class TestContainerDiscoverer : ITestContainerDiscoverer, IDisposable {
 
         public IEnumerable<ITestContainer> TestContainers {
             get {
-                if (IsSolutionMode()
-                    && (_firstLoad || _isRefresh)) {
-                    _projectMap.Clear();
-
+                if (IsSolutionMode() && 
+                    (_firstLoad || _isRefresh)) {
                     // The first time through, we don't know about any loaded
                     // projects.
                     ThreadHelper.JoinableTaskFactory.Run(async () => {
                         await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
                         if (_firstLoad || _isRefresh) {
-                            SetupSolution();
                             _firstLoad = false;
                             _isRefresh = false;
+                            
```

Comment:
safer to reset these before calling the functions below.  this all runs in SwitchToMainThreadAsync so that after allowing another refresh it will still have to wait to SetupSolution() to complete